kafka_plugin.cpp

```diff
@@ -453,14 +491,25 @@ using kafka_producer_ptr = std::shared_ptr<class kafka_producer>;
        producer->trx_kafka_sendmsg(KAFKA_TRX_APPLIED,
                                    (char*)transaction_metadata_json.c_str(),
                                    sstream.str());
-
-       if(0 == t.block_number % KAFKA_BLOCK_REACHED_LOG_INTERVAL &&
-               kafkaBlockReached < t.block_number) {
+       if(0 == kafkaBlockReached) {
            kafkaBlockReached = t.block_number;
-           ilog("Block number ${block_number} reached Kafka plugin",
-                ("block_number", t.block_number));
+           prometheusExposer->getBlockCounter().Increment(kafkaBlockReached);
+       }
+       else if(kafkaBlockReached > t.block_number) {
+           // Mini fork detected
+           prometheusExposer->getAbnormalityBlockCounter().Increment(kafkaBlockReached - abnormalityDetectedAtBlock);
+           abnormalityDetectedAtBlock = kafkaBlockReached;
+       } else if( t.block_number > kafkaBlockReached + 1) {
+           // Jump over blocks detected. This should not really happen.
+           prometheusExposer->getAbnormalityBlockCounter().Increment(kafkaBlockReached - abnormalityDetectedAtBlock);
+           abnormalityDetectedAtBlock = kafkaBlockReached;
+           prometheusExposer->getBlockCounter().Increment(t.block_number - kafkaBlockReached);
+           kafkaBlockReached = t.block_number;
+       } else if(t.block_number == kafkaBlockReached + 1){
+           // Normal case.
+           prometheusExposer->getBlockCounter().Increment();
+           ++kafkaBlockReached;
```

Comment:
I think if a gauge is used most probably it will be possible to remove this last condition and just have a single line at the beginning doing:

```cpp
prometheusExposer->getBlockGauge().Set(t.block_number);
```

Reply:
I env think that the variables `kafkaBlockReached` and `abnormalityDetectedAtBlock` can be removed.

This code should be possible to be reduces to just:

```cpp
if (prometheusExposer->getBlockGauge().Value()) {
  prometheusExposer->getAbnormalityBlockGauge(t.block_number)
}
prometheusExposer->getBlockGauge().Set(t.block_number);
```

The above is missing the case for the block skips, but I think it can be included easily. May be with an initial step to check if the counter is 0 at the moment.

Reply:
Indeed I had removed the internally used variables.